app/header.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  View,
+  Text,
+  ColorValue,
+  StyleProp,
+  TextStyle,
+  StyleSheet,
+} from "react-native";
+import Button from "../components/Button";
+import {
+  Ionicons,
+  Entypo,
+  MaterialCommunityIcons,
+  FontAwesome5,
+} from "@expo/vector-icons";
+import { styled, useSx } from "dripsy";
+
+interface HeaderProps {
+  title: string;
+  titleColor?: ColorValue;
+  style?: StyleProp<TextStyle> | undefined;
+  search?: boolean;
+  searchOnPress?: () => void;
+  messenger?: boolean;
+  messengerOnPress?: () => void;
+  plus?: boolean;
+  plusOnPress?: () => void;
+  settings?: boolean;
+  settingsOnPress?: () => void;
+  userData?: boolean;
+  userDataOnPress?: () => void;
+}
+
+const styles = StyleSheet.create({

```

Comment:
define this to the button component, add a type property for this kind of style of a button